### PR TITLE
Add GethMiner snake case methods

### PR DESCRIPTION
**Pull request by billbsing**
### What was wrong?

Related to Issue #1429

### How was it fixed?
add new methods, and mark the current non snake case method names as depreciated.

### CODE REVIEW

File: tests/core/mining-module/test_miner_setGasPrice.py [hunk 1]
```diff
@@ -10,15 +11,35 @@
...
     initial_gas_price = web3.eth.gasPrice
 
     # sanity check
     assert web3.eth.gasPrice > 1000
 
-    web3.geth.miner.setGasPrice(initial_gas_price // 2)
+    web3.geth.miner.set_gas_price(initial_gas_price // 2)
+
+    with Timeout(60) as timeout:
+        while web3.eth.gasPrice == initial_gas_price:
```
Comment (kclowes) [inline tests/core/mining-module/test_miner_setGasPrice.py]: even tough I'm pretty sure these are also getting skipped, let's change `gasPrice` to `gas_price`
```suggestion
        while web3.eth.gas_price == initial_gas_price:
```

File: docs/web3.miner.rst [hunk 2]
```diff
@@ -71,23 +88,33 @@ The following methods are available on the ``web3.geth.miner`` namespace.
...
     * Delegates to ``miner_startAutoDag`` RPC Method
 
     Enable automatic DAG generation.
 
     .. code-block:: python
 
-        >>> web3.geth.miner.startAutoDAG()
+        >>> web3.geth.miner.start_auto_dag()
 
+.. py:method:: GethMiner.startAutoDAG()
```
Comment (kclowes) [inline docs/web3.miner.rst]: This is existing, but we only have a `startAutoDag` method, not `startAutoDAG`. Do you mind changing?
```suggestion
.. py:method:: GethMiner.startAutoDag()
```

File: docs/web3.miner.rst [hunk 3]
```diff
@@ -71,23 +88,33 @@ The following methods are available on the ``web3.geth.miner`` namespace.
...
     * Delegates to ``miner_stopAutoDag`` RPC Method
 
     Disable automatic DAG generation.
 
     .. code-block:: python
 
-        >>> web3.geth.miner.stopAutoDAG()
+        >>> web3.geth.miner.stop_auto_dag()
+
+.. py:method:: GethMiner.stopAutoDAG()
```
Comment (kclowes) [inline docs/web3.miner.rst]: Same here
```suggestion
.. py:method:: GethMiner.stopAutoDag()
```

File: tests/core/mining-module/test_miner_setGasPrice.py (hunk 1 above)
Comment (billbsing) [reply tests/core/mining-module/test_miner_setGasPrice.py]: Not sure I should be doing this in this PR, the property `gasPrice` is from the 'eth' module. I'm happy to do the eth module later..

File: tests/core/mining-module/test_setEtherBase.py [hunk 4]
```diff
@@ -1,7 +1,18 @@
+import pytest
+
+
+def test_miner_set_etherbase(web3_empty):
+    web3 = web3_empty
+    assert web3.eth.coinbase == web3.eth.accounts[0]
+    new_account = web3.personal.newAccount('this-is-a-password')
+    web3.geth.miner.set_etherBase(new_account)
```
[the remark this answers is not in the document]
Comment (billbsing) [reply tests/core/mining-module/test_setEtherBase.py]: thanks missed that one.. in theory we should also rename the test file, to `test_setEtherbase.py', but I'll leave it later to rename file names correctly to the snake case version 'test_set_etherbase.py'

File: tests/core/mining-module/test_miner_setGasPrice.py (hunk 1 above)
Comment (kclowes) [reply tests/core/mining-module/test_miner_setGasPrice.py]: Ha, you're right. My bad!

**Pull request merged**